bluepysnap/nodes.py

```diff
@@ -27,103 +27,35 @@
-                    raise BluepySnapError("Duplicated node population: '%s'" % population)
-                res[population] = storage.population(population)
-        return res
-
-    @cached_property
-    def population_names(self):
-        """Returns all the population names from the Circuit."""
-        return sorted(self._populations)
+        super().__init__(circuit)
 
-    @cached_property
-    def property_dtypes(self):
-        """Returns all the property dtypes for the Circuit."""
-        def _update(d, index, value):
-            if d.setdefault(index, value) != value:
-                raise BluepySnapError("Same property with different "
-                                      "dtype. {}: {}!= {}".format(index, value, d[index]))
-
-        res = dict()
-        for pop in self.values():
-            for varname, dtype in pop.property_dtypes.iteritems():
-                _update(res, varname, dtype)
-        return pd.Series(res)
-
-    def keys(self):
-        """Returns iterator on the population names.
-
-        Made to simulate the behavior of a dict.keys().
-        """
-        return (name for name in self.population_names)
-
-    def values(self):
-        """Returns iterator on the NodePopulations.
-
-        Made to simulate the behavior of a dict.values().
-        """
-        return (self[name] for name in self.population_names)
-
-    def items(self):
-        """Returns iterator on the tuples (population name, NodePopulations).
-
-        Made to simulate the behavior of a dict.items().
-        """
-        return ((name, self[name]) for name in self.population_names)
-
-    def __getitem__(self, population_name):
-        """Access the NodePopulation corresponding to the population 'population_name'."""
-        try:
-            return self._populations[population_name]
-        except KeyError:
-            raise BluepySnapError("{} not a node population.".format(population_name))
-
-    def __iter__(self):
-        """Allows iteration over the different NodePopulation."""
-        return iter(self.keys())
-
-    @cached_property
-    def size(self):
-        """Total number of nodes inside the circuit."""
-        return sum(pop.size for pop in self.values())
-
-    @cached_property
-    def property_names(self):
-        """Returns all the properties present inside the circuit."""
-        return set(prop for pop in self.values() for prop in pop.property_names)
+    def _collect_populations(self):
+        return self._get_populations(NodeStorage, self._config['networks']['nodes'])
 
     def property_values(self, prop):
-        """Returns all the values for a given property."""
+        """Returns all the values for a given Nodes property."""
         return set(value for pop in self.values() if prop in pop.property_names for value in
                    pop.property_values(prop))
 
-    def ids(self, group=None):
+    def ids(self, group=None):   # pylint: disable=arguments-differ
```

Comment:
Just FYI: The rationale they give for this warning is pretty good.
https://github.com/PyCQA/pylint/issues/3638

Reply:
This is something I would like to change when the properties/get from the edges remove the possibility to return the edge_ids. 
Once this is done, I will change the signatures accordingly.